src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -231,7 +232,40 @@ internal static byte[] Encode(CoseHeaderMap? map, bool mustReturnEmptyBstrIfEmpt
                 writer.WriteEncodedValue(encodedValue.Span);
             }
             writer.WriteEndMap();
-            return writer.Encode();
+
+            return writer.Encode(destination);
+        }
+
+        internal static int ComputeEncodedSize(CoseHeaderMap? map, bool isProtectedHeader, int? algHeaderValueToSlip = null)
+        {
+            map ??= s_emptyMap;
+
+            // encoded map length => map length + (label + value)*
+            Debug.Assert(algHeaderValueToSlip == null || isProtectedHeader);
+
+            int encodedSize = 0;
+            int mapLength = map._headerParameters.Count;
+            if (algHeaderValueToSlip != null)
+            {
+                mapLength += 1;
+                encodedSize += CoseHeaderLabel.Algorithm.EncodedSize;
+                encodedSize += CoseHelpers.GetEncodedSize(algHeaderValueToSlip.Value);
+            }
+
+            encodedSize += CoseHelpers.GetEncodedSize(mapLength);
+
+            foreach ((CoseHeaderLabel label, ReadOnlyMemory<byte> encodedValue) in map)
+            {
+                encodedSize += label.EncodedSize + encodedValue.Length;
+            }
+
+            // bstr(encoded map length)
+            if (isProtectedHeader)
+            {
+                encodedSize = CoseHelpers.GetEncodedSize(encodedSize) + encodedSize;
```

Comment:
```suggestion
                encodedSize += CoseHelpers.GetEncodedSize(encodedSize);
```

Everything else was +=, it was weird that this one was the same logic but a different style.

Reply:
The encoded size of a bstr is `encoded size of the bytes length + bytes length`, that's why I felt this way was clearer.